### PR TITLE
extension: use bool_flag to control extension link

### PR DESCRIPTION
**Pull request by lizan**
Pulling bazel stuff out of #14094. Use the new kill request extension as example.
 
Signed-off-by: Lizan Zhou <lizan@tetrate.io>

### CODE REVIEW

File: test/extensions/extensions_build_system.bzl [hunk 1]
```diff
@@ -1,62 +1,47 @@
 load("//bazel:envoy_build_system.bzl", "envoy_benchmark_test", "envoy_cc_benchmark_binary", "envoy_cc_mock", "envoy_cc_test", "envoy_cc_test_binary", "envoy_cc_test_library")
-load("@envoy_build_config//:extensions_build_config.bzl", "DISABLED_BY_DEFAULT_EXTENSIONS", "EXTENSIONS")
+load("@envoy_build_config//:extensions_build_config.bzl", "EXTENSIONS")
 
-def extension_allowed_for_test(extension_name):
-    return extension_name in EXTENSIONS or extension_name in DISABLED_BY_DEFAULT_EXTENSIONS
+def _skip_extension(func, name, extension_name, **kwargs):
```
Comment (htuch) [inline test/extensions/extensions_build_system.bzl:4]: This name is a bit confusing, maybe `apply_to_extension_allow_for_test` or something.

File: source/extensions/extensions_build_config.bzl [hunk 2]
```diff
@@ -71,9 +71,11 @@ EXTENSIONS = {
     "envoy.filters.http.health_check":                  "//source/extensions/filters/http/health_check:config",
     "envoy.filters.http.ip_tagging":                    "//source/extensions/filters/http/ip_tagging:config",
     "envoy.filters.http.jwt_authn":                     "//source/extensions/filters/http/jwt_authn:config",
+    # Disabled by default
```
Comment (htuch) [inline source/extensions/extensions_build_config.bzl:74]: Can we have some test, maybe extending https://github.com/envoyproxy/envoy/blob/master/test/exe/main_common_test.cc#L459 or even just use `bazel query` that verifies that these aren't linked by default?

**Review by htuch — CHANGES_REQUESTED** [review]
Overall makes sense, this is really nice. 
/wait

File: source/extensions/extensions_build_config.bzl (hunk 2 above)
Comment (lizan) [reply source/extensions/extensions_build_config.bzl:74]: https://github.com/envoyproxy/envoy/blob/master/test/exe/main_common_test.cc#L459 already covers, and it passes. Or do you want to make that test pass even with the command line flag?

Comment (htuch) [reply source/extensions/extensions_build_config.bzl:74]: I guess my main reservation is that we probably should have an inclusion/exclusion CLI test that verifies the effects here. The reason I'm being picky is that for some sites, it's security critical to ensure that some extension are definitely excluded.

Comment (lizan) [reply source/extensions/extensions_build_config.bzl:74]: ok added inclusion CLI test in compile time options, exclusion will be part of wasm runtime enable/disable PR.

**Review by htuch — APPROVED** [review]
LGTM, thanks!

**Pull request merged**